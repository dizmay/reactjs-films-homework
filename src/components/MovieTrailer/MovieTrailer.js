import React from 'react';
import PropTypes from 'prop-types';
import styles from './MovieTrailer.scss';

const MovieTrailer = ({ movieTrailer, showTrailer }) => (
  <div className={styles.trailer__container}>
    <button type="button" onClick={showTrailer} className={styles.trailer__close}>x</button>
    <div className={styles.trailer__modal}>
      <iframe className={styles.trailer__video} src={`https://www.youtube.com/embed/${movieTrailer}`} title="Movie trailer" />
    </div>
  </div>
);

MovieTrailer.defaultProps = {
  movieTrailer: '',
  showTrailer: null,
};

MovieTrailer.propTypes = {
  movieTrailer: PropTypes.string,
  showTrailer: PropTypes.func,
};

export default MovieTrailer;
