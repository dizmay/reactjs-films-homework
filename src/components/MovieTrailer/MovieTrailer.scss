* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.trailer__container {
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100vw;
  height: 100vh;
  background: #000000b3;
  z-index: 999;
}

.trailer__close {
  position: absolute;
  top: 2rem;
  right: 2rem;
  background: transparent;
  border: none;
  outline: none;
  color: #fff;
  font-size: 4rem;
  cursor: pointer;
}

.trailer__modal {
  width: 60%;
  height: 60%;
  background: #000;
}

.trailer__video {
  width: 100%;
  height: 100%;
  border: 0.2rem solid #00a3f5;
  object-fit: cover;
}
