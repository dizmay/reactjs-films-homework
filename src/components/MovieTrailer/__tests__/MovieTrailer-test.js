import React from 'react';
import renderer from 'react-test-renderer';
import MovieTrailer from '../MovieTrailer';

it('renders correctly', () => {
  const tree = renderer.create(<MovieTrailer />).toJSON();
  expect(tree).toMatchSnapshot();
});
