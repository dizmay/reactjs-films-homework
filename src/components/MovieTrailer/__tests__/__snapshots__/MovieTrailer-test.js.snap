// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="trailer__container"
>
  <button
    className="trailer__close"
    onClick={null}
    type="button"
  >
    x
  </button>
  <div
    className="trailer__modal"
  >
    <iframe
      className="trailer__video"
      src="https://www.youtube.com/embed/"
      title="Movie trailer"
    />
  </div>
</div>
`;
